%*******************************************************************************
%****************************** Second Chapter *********************************
%*******************************************************************************

\chapter{Conclusions and Future work}

\ifpdf
    \graphicspath{{Chapter6/Figs/Raster/}{Chapter6/Figs/PDF/}{Chapter6/Figs/}}
\else
    \graphicspath{{Chapter6/Figs/Vector/}{Chapter6/Figs/}}
\fi

The final chapter of this thesis aim to summarise the key ideas and main contributions of the previous chapters and draw some attentions for future work.

\section{Summary of the Contributions}
\textbf{Chapter 2:} We have reimplemented the tool \emph{chipDyno} using \emph{R} programming language with the aim to make it public through a open source platform GitHub. We constructed our connectivity information between genes and transcription factors from the evidence of gene to gene interaction to model the TFAs. Earlier the dynamics of TFAs were obtained for a unicellular microorganism (yeast), our tool modeled transcription factor activities for a multicellular eukaryote (\textit{C.elegans}). The probabilistic dynamical model for quantitative inference of TFA, that we described in this chapter used as the basis for the  Chapter \ref{ch:GaussianProcessRegression} and Chapter \ref{ch:GP_Model_of_TFAs}.

\textbf{Chapter 3:} In this technical background chapter we briefly described Gaussian process, regression problem and regression with Gaussian process. The choice of the covariance function is a central step in modelling with a Gaussian process. We justified the rationale behind choosing the Ornstein-Uhlenbeck kernel to model the transcription factor activity using Gaussian process. Then we developed a special covariance functions suitable for transcription factor activity analysis. 

\textbf{Chapter 4:} The linear Gaussian model is equivalent to a Gaussian process with a particular covariance function. We therefore build a model directly from the Gaussian process perspective. Here we designed a covariance function for reconstructing transcription factor activities given gene expression profiles and a connectivity information between genes and transcription factors. The joint process across all transcription factor activities and across all time points might have some correlation, here we incorporated intrinsic model of coregionalization for the the joint process. We also introduced a computational trick, based on  judicious application of singular value decomposition, to enable us to efficiently fit the Gaussian process in a reduced \lq TF activity\rq space. 

\textbf{Chapter 5:} We have performed genome-wide analysis to cluster genes systematically and analyse the rationale behind the variation in the speed of propagation for ALS. Our particular innovation was to include the condition and genetic background of the organisms within the underlying functional component of our clusters. This ensured that the underlying expressions behaved similarly were more likely to be clustered together. We used a widely acceptable gene ontology and functional annotation tool to validate our clusters and their characteristics obtained from our model. Gene expression time series characteristics curve and enrichment scores analyse helped us to narrow down our search and lead toward finding the lists of genes or clusters which could be involved in the speed of disease propagation. Our pathway analysis found evidence of genes which are known to be involved in the disease process. The special covariance function we have developed for clustering considering models condition, genetic background, replicates and disease states with coregionalization could be useful to investigate other biological activity where clustering is required. 

\section{Future Work}
\textbf{Bridge between TFA and clustering:} ?


\textbf{Validation using Federico's model:} ?


\textbf{Deep learning:} ?


\textbf{Big Data:} ?


\textbf{Application:} ?

% \cite{Sanguinetti:2006} model to infer the transcription factor 
% activity is a linear- Gaussian state-space model. We believe that this linear Gaussian
% model is equivalent to Gaussian process with a specific covariance function.
% We have developed a model directly from Gaussian process to achieve the same goal.
% We are quite close to develop a valid covariance function for reconstructing transcription
% factor activities given gene expression profile and binding information between genes and
% transcription factors. Here we will introduce a computational trick using
% singular value decomposition and intrinsic coregionalization model. We believe
% this method will enable us to efficiently fit the Gaussian process in a
% reduced transcription factor activity space.
% 
% Amyotrophic lateral sclerosis (ALS), also known as ``Lou Gehrig's Disease'' or motor neurone disease 
% (MND), is an irreversible progressive neurodegenerative adult onset that affects motor neurons 
% in the brain and the spinal cord. Muscle denervation spreads over neuromuscular system and 
% leads toward death by failure of the respiratory system with in few years of system 
% onset (\cite{Peviani:2010}). 
% This lethal invariable disorder has median survival of less than 5 years, only 20\% 
% of the affected people can survive more than 5 years and 10\% of the patients 
% can survive more than 10 years. Mutation in the Cu/Zn superoxide dismutase (SOD1) gene 
% is responsible for around 20\% of the familial motor neurone disease \cite{Nardo:2013}. 
% Transgenic mice can express human SOD1 mutation and nicely replicates different 
% histopathological and clinical features of motor neurone disease. Mimic of these murine models 
% of different clinical phenotypes observed from human MND patients are widely using by the 
% researchers to determine the disease progression. But we didn't found any evidence of gene expression 
% analysis that attempt to analyse motor neuron disease considering the genetic background on 
% different phenotype of this murine disease. So gene expression analysis for different murine models 
% could reveal interesting information. 
% \cite{Nardo:2013} used two mouse models to analyse fast and slow disease progression of ALS.
% We will use our Gaussian process based model to infer the 
% transcription factor activity on gene expression data obtained from different murine models and 
% try to find out some fascinating insights.
% 
% Clustering of gene expression time series is another major interest of the research to get the view of
% groups of co-regulated or associated genes. It is assumed that gene involved in the same biological 
% process will be expressed with a similarity sharing underlying time series. \cite{Cossins:2007}
% did some additional cluster analysis (not published yet!) based on some phenotype properties. Again 
% it is very common to have multiple biological replicates of the gene expression time series data. 
% Just taking average of the replicates surely lead toward discarding insight. Recently \cite{Hensman:2013} used
% a hierarchy of Gaussian process to model a gene specific and replicate specific temporal covariance.
% They also used this model for clustering application. Using this Gaussian process based hierarchical
% clustering analysis of \cite{Hensman:2013} we will try to find some robust clusters for the gene
% expression data of \textit{C. elegans}. Once if we can do so, it will easily lead us to find out the active
% transcription factors related with these clusters and their subsequent dynamic behaviour as well.